runtime/v2/manager.go

```diff
@@ -104,49 +160,54 @@ type ManagerConfig struct {
 	SchedCore    bool
 }
 
-// New task manager for v2 shims
-func New(ctx context.Context, config *ManagerConfig) (*TaskManager, error) {
+// NewShimManager creates a manager for v2 shims
+func NewShimManager(ctx context.Context, config *ManagerConfig) (*ShimManager, error) {
 	for _, d := range []string{config.Root, config.State} {
 		if err := os.MkdirAll(d, 0711); err != nil {
 			return nil, err
 		}
 	}
-	m := &TaskManager{
+
+	m := &ShimManager{
 		root:                   config.Root,
 		state:                  config.State,
 		containerdAddress:      config.Address,
 		containerdTTRPCAddress: config.TTRPCAddress,
-		schedCore:              config.SchedCore,
-		tasks:                  runtime.NewTaskList(),
+		list:                   runtime.NewTaskList(),
 		events:                 config.Events,
 		containers:             config.Store,
+		schedCore:              config.SchedCore,
 	}
+
 	if err := m.loadExistingTasks(ctx); err != nil {
 		return nil, err
 	}
+
 	return m, nil
 }
 
-// TaskManager manages v2 shim's and their tasks
-type TaskManager struct {
+// ShimManager manages currently running shim processes.
+// It is mainly responsible for launching new shims and for proper shutdown and cleanup of existing instances.
+// The manager is unaware of the underlying services shim provides and lets higher level services consume them,
+// but don't care about lifecycle management.
+type ShimManager struct {
 	root                   string
 	state                  string
 	containerdAddress      string
 	containerdTTRPCAddress string
 	schedCore              bool
-
-	tasks      *runtime.TaskList
-	events     *exchange.Exchange
-	containers containers.Store
+	list                   *runtime.TaskList
```

Comment:
I prefer `tasks` over `list` since it is less abstract.

Reply:
I'm not sure about `tasks` because `shim service != task service`. and it's not exactly tasks, but shim instances.

Reply:
Oh I was assuming so because of the type. Could it be `shims` then?